src/NHibernate/Linq/Assignments.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using NHibernate.Linq.Visitors;
+using Remotion.Linq.Utilities;
+
+namespace NHibernate.Linq
+{
+	public abstract class Assignments
+	{
+		protected static readonly ConstructorInfo DictionaryConstructorInfo = typeof(Dictionary<string, object>).GetConstructor(new[] { typeof(int) });
+		protected static readonly MethodInfo DictionaryAddMethodInfo = typeof(Dictionary<string, object>).GetMethod("Add");
+	}
+
+	/// <summary>
+	/// Class to hold assigments used in updates and inserts
+	/// </summary>
+	/// <typeparam name="TInput">The type of the input.</typeparam>
+	/// <typeparam name="TOutput">The type of the output.</typeparam>
+	public class Assignments<TInput, TOutput> : Assignments
+	{
+		private readonly List<Assignment> _sets = new List<Assignment>();
+
+		/// <summary>
+		/// Sets the specified property.
+		/// </summary>
+		/// <typeparam name="TProp">The type of the property.</typeparam>
+		/// <param name="property">The property.</param>
+		/// <param name="expression">The expression that should be assigned to the property.</param>
+		/// <returns></returns>
+		public Assignments<TInput, TOutput> Set<TProp>(Expression<Func<TOutput, TProp>> property, Expression<Func<TInput, TProp>> expression)
+		{
+			ArgumentUtility.CheckNotNull("property", property);
+			ArgumentUtility.CheckNotNull("expression", expression);
```

Comment:
This utility seems to be no more available after rebasing, I have re-coded it "old style" on my local branch.